File: Python/Harddrive.py
from Komponen import Komponen

class Harddrive(Komponen):
    __kapasitasGB = 0
    __tipeDrive = ""

    def __init__(self, kapasitasGB, tipeDrive, merk, nama):
        super().__init__(merk, nama)
        self.__kapasitasGB = kapasitasGB
        self.__tipeDrive = tipeDrive

    def get_kapasitasGB(self):
        return self.__kapasitasGB
    
    def get_tipeDrive(self):
        return self.__tipeDrive
    
    def set_kapasitasGB(self, kapasitasGB):
        self.__kapasitasGB = kapasitasGB

    def set_tipeDrive(self, tipeDrive):
        self.__tipeDrive = tipeDrive

File: Java/Harddrive.java
public class Harddrive extends Komponen
{
    private int kapasitasGB;
    private String tipeDrive;

    public Harddrive()
    {
        this.kapasitasGB = 0;
        this.tipeDrive = "";
    }

    public Harddrive(int kapasitasGB, String tipeDrive, String merk, String nama)
    {   
        super(merk, nama);
        this.kapasitasGB = kapasitasGB;
        this.tipeDrive = tipeDrive;
    }

    public void set_kapasitasGB(int kapasitasGB)
    {
        this.kapasitasGB = kapasitasGB;
    }

    public void set_tipeDrive(String tipeDrive)
    {
        this.tipeDrive = tipeDrive;
    }

    public int get_kapasitasGB()
    {
        return this.kapasitasGB;
    }

    public String get_tipeDrive()
    {
        return this.tipeDrive;
    }
}